lib/lib-utils/src/main/java/org/sagebionetworks/util/json/JavaJSONUtil.java

```diff
@@ -84,6 +94,84 @@ public static <T> List<T> readFromJSON(Class<? extends T> clazz, JSONArray array
 		return list;
 	}
 
+	/**
+	 * Read a list of objects directly from the provided Stream
+	 * @param <T>
+	 * @param clazz
+	 * @param reader
+	 * @return
+	 */
+	public static <T> List<T> readFromJSON(Class<? extends T> clazz, Reader reader) {
+		ValidateArgument.required(reader, "reader");
+		ValidateArgument.required(clazz, "clazz");
+		Map<String, Object> typeMap = Arrays.stream(clazz.getDeclaredFields())
+				.collect(Collectors.toMap(f -> f.getName(), f -> f.getType()));
+		try {
+			List<T> list = new ArrayList<>();
+			try (JsonReader stream = new JsonReader(reader)) {
+				stream.beginArray();
+				while (stream.hasNext()) {
+					list.add(readFromJSON(clazz, readObject(typeMap, stream)));
+				}
+				stream.endArray();
+			}
+			return list;
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Read a single JSONObject from the provided stream.
+	 * 
+	 * @param stream
+	 * @return
+	 * @throws IOException
+	 */
+	public static JSONObject readObject(Map<String, Object> typeMap, JsonReader stream) throws IOException {
```

Comment:
I think this would not work if there were nested objects, I would simply use the JSONTokener (See previous comment)